yt/yt/server/node/exec_node/slot_manager.cpp

```diff
@@ -247,6 +242,25 @@ TFuture<void> TSlotManager::InitializeEnvironment()
         Bootstrap_->GetConfig()->JobResourceManager->ResourceLimits->Cpu,
         GetIdleCpuFraction());
 
+    auto slotInitTimeout = DynamicConfig_.Acquire()->SlotInitTimeout;
+    for (int slotIndex = 0; slotIndex < SlotCount_; ++slotIndex) {
+        auto slotInitFuture = JobEnvironment_->InitSlot(slotIndex)
+            .WithTimeout(slotInitTimeout);
+        slotInitFuture.Subscribe(
+            BIND([this, this_ = MakeStrong(this), slotIndex] (const TError& error) {
+                VERIFY_THREAD_AFFINITY(JobThread);
+
+                if (error.IsOK()) {
+                    FreeSlots_.push(slotIndex);
+                } else {
+                    FreeSlots_.clear();
```

Comment:
Why do you do it here? This is not the only place where job env can be disabled. I'd suggest to clean free slots just before this loop.